build/marketplace/MarketPlace.sol

```diff
@@ -188,11 +205,111 @@ contract MarketPlace {
         uint256 m,
         uint128 a
     ) external unpaused(p) returns (uint128) {
-        IPool pool = IPool(pools[u][m][p]);
+        IPool pool = IPool(pools[u][m]);
         Safe.transfer(IErc20(address(pool.principalToken())), address(pool), a);
         return pool.buyUnderlying(msg.sender, pool.buyUnderlyingPreview(a), a);
     }
 
+    /// @dev Mint liquidity tokens in exchange for adding underlying and PT.
+    /// The amount of liquidity tokens to mint is calculated from the amount of
+    /// unaccounted for PT in this contract.
+    /// A proportional amount of underlying tokens need to be present in this
+    /// contract, also unaccounted for.
+    /// @param u the address of the underlying token
+    /// @param m the maturity of the principal token
+    /// @param underlyingAmount the underlying amount being sent
+    /// @param principalTokenAmount the principal token amount being sent
+    /// @param minRatio Minimum ratio of underlying to PT in the pool.
+    /// @param maxRatio Maximum ratio of underlying to PT in the pool.
+    /// @return The amount of liquidity tokens minted.
+    function mint(
+        address u,
+        uint256 m,
+        uint256 underlyingAmount,
+        uint256 principalTokenAmount,
+        uint256 minRatio,
+        uint256 maxRatio
+    )
+        external
```

Comment:
inline the external, modifiers, and returns (just keeping format with our other methods)